nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,674 @@
+                 smoothing_fwhm=None, standardize=True, target_affine=None,
+                 target_shape=None, mask_strategy='background',
+                 memory=None, memory_level=0,
+                 n_jobs=1, verbose=False):
+        self.estimator = estimator
+        self.mask = mask
+        self.cv = cv
+        self.param_grid = param_grid
+        self.screening_percentile = screening_percentile
+        self.pos_label = pos_label
+        self.scoring = scoring
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def _gather_fit_results(self, results, classes_to_predict, score_func):
+        """Posprocessing of the results"""
+        coefs = {}
+        intercepts = {}
+        cv_true = {}
+        self.cv_params_ = {}
+
+        cv_pred = []
+        cv_true = []
+        for c, best_coef, best_intercept, best_y, best_params in results:
+            cv_pred_ = {}
+            cv_true_ = {}
+            cv_pred_[c] = best_y['y_pred']
+            cv_true_[c] = best_y['y_true']
+
+            coefs.setdefault(c, []).append(best_coef)
+            intercepts.setdefault(c, []).append(best_intercept)
+            self.cv_params_.setdefault(c, {})
+            for k in best_params:
+                self.cv_params_[c].setdefault(k, []).append(best_params[k])
+
+            if self.is_binary_:
+                other_class = np.setdiff1d(
+                    self.classes_, classes_to_predict)[0]
+                coefs.setdefault(other_class, []).append(-best_coef)
+                intercepts.setdefault(other_class, []).append(-best_intercept)
+                cv_pred_[other_class] = best_y['inverse']
+
+            cv_pred.append(cv_pred_)
+            cv_true.append(cv_true_)
+
+        if self.is_classification_:
+            classes = self.classes_
+        else:
+            classes = classes_to_predict
+
+        self.cv_y_pred_ = []
+        self.cv_y_true_ = []
+        self.cv_scores_ = []
+
+        # For each cv loop
+        for k in range(len(cv_pred)):
+            # if is classification this corresponds to y_prob
+            y_pred = np.vstack([cv_pred[k][c] for c in classes]).T
+            y_true = cv_true[k][list(cv_true[k].keys())[0]]
+
+            if self.is_classification_:
+                y_pred = self.classes_[np.argmax(y_pred, axis=1)]
+
+            self.cv_y_true_.append(y_true)
+            self.cv_y_pred_.append(y_pred)
+            self.cv_scores_.append(score_func(y_true, y_pred))
+
+        self.coef_ = np.vstack([np.mean(coefs[c], axis=0) for c in classes])
+        self.std_coef_ = np.vstack([np.std(coefs[c], axis=0) for c in classes])
```

Comment:
Let us not store coef_ and std_coef_ in the estimator as we are storing the images in coef_img_ and std_coef_img_